lib/trivial.nix

```
@@ -12,7 +12,20 @@ let
     version
     versionSuffix
     warn;
-in {
+
+  # We have to use builtins here in place of lib.warn, as otherwise its
+  # definition would depend on itself. Further, we do not want this to
+  # actually abort even if NIX_ABORT_ON_WARN is set, as otherwise legacy
+  # uses of this variable would immediately result in an unrecoverable
+  # error where there should be a deprecation warning.
+  printDeprecationWarning =
+    if builtins.getEnv "NIX_ABORT_ON_WARN" != "" then
```

```suggestion
    if lib.isInOldestRelease 2405 && builtins.getEnv "NIX_ABORT_ON_WARN" != "" then
```

This needs some code rearranging to get working without infinite recursion, but this makes sure that there's a transition period where stable third party projects can switch to `NIXPKGS_ABORT_ON_WARN` without warnings for end-users.

---

I have done so, though aesthetically I don't particularly like the result. If you can think of any nicer way to do this, please let me know / feel free to edit this branch.